tests/unittests/MLTest.cpp

```diff
@@ -205,6 +207,79 @@ TEST_P(MLTest, learnXor) {
   }
 }
 
+/// Learn the logarithmic function
+TEST_P(MLTest, learnLog) {
+  unsigned numInputs = 500;
+  unsigned batchSize = 10;
+  EE_.getConfig().learningRate = 0.05;
+  EE_.getConfig().batchSize = batchSize;
+
+  auto &mod = EE_.getModule();
+  Function *F = mod.createFunction("main");
+  F->setName("learnLog");
+
+  auto *A =
+      mod.createVariable(ElemKind::FloatTy, {batchSize, 1}, "A", VisibilityKind::Public,
+                         Variable::TrainKind::None);
+  auto *Ex =
+      mod.createVariable(ElemKind::FloatTy, {batchSize, 1}, "Ex", VisibilityKind::Public,
+                         Variable::TrainKind::None);
+
+  Node *O = F->createFullyConnected("fc1", A, 5);
```

Comment:
Why not use `auto` here too?

Reply:
because i use the same variable O here for all the node-creating function. I have to set the type of O as the Node, not to be any specific subclass class.